test/browsing.js

```diff
@@ -81,171 +80,143 @@ const removeFromMData = (md, key) => md.getEntries()
         )));
 
 describe('Browsing', () => {
+  let app;
+
+  before(async () => {
+    const containersPermissions = { _public: ['Read'], _publicNames: ['Read', 'Insert', 'ManagePermissions'] };
+    app = await createAuthenticatedTestApp('_test_scope', containersPermissions);
+  });
+
   it('returns rejected promise if no url is provided', () => {
     const content = `hello world, on ${Math.round(Math.random() * 100000)}`;
-    return createRandomDomain(content, '', '')
-      .then(() => createAnonTestApp()
-        .then((app) => should(app.webFetch()).be.rejectedWith(errConst.MISSING_URL.msg)));
+    return createRandomDomain(content, '', '', app)
+      .then(() => should(app.webFetch()).be.rejectedWith(errConst.MISSING_URL.msg));
```

Comment:
Strictly speaking what you are doing here is not exactly the same, since you are using a single authenticated app for both publishing the site and for fetching it with `webFetch`, while the original code is using a separate un-authenticated app for webFetching the site, which is what happens when you are browsing sites.

Since the enhancement you are doing is really good as you moved the creation of the authenticated app to the `before` body, you can simply add a the creation of a second and un-authenticated app in the `before` as well, and it shouldn't impact the performance you gained with these changes.